File: AFMobile/AFAndroid/app/src/main/java/cz/cvut/fel/matyapav/afandroid/components/parts/validators/MinValueValidator.java
package cz.cvut.fel.matyapav.afandroid.components.parts.validators;

import android.widget.EditText;

import com.tomscz.afrest.rest.dto.AFValidationRule;

import cz.cvut.fel.matyapav.afandroid.components.parts.AFField;
import cz.cvut.fel.matyapav.afandroid.utils.Localization;
import cz.cvut.fel.matyapav.afandroid.utils.Utils;

/**
 * Validates MIN rule, which checks if value in field is more or equal than value specified in rule.
 *
 * @author Pavel Matyáš (matyapav@fel.cvut.cz)
 *
 * @since 1.0.0.
 */
public class MinValueValidator implements AFValidator {

    @Override
    public boolean validate(AFField field, StringBuilder errorMsgs, AFValidationRule rule) {
        boolean validationIsFine = true;
        if(Utils.isFieldNumberField(field)){
            EditText numberField = (EditText) field.getFieldView();
            if(numberField.getText() != null && !numberField.getText().toString().isEmpty() &&
                    Double.valueOf(numberField.getText().toString()) < Double.valueOf(rule.getValue())){
                validationIsFine = false;
            }
        }
        if(!validationIsFine){
            errorMsgs.append(Localization.translate("validation.minval")+" "+rule.getValue());
        }
        return validationIsFine;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/components/parts/validators/MinValueValidator.cs
using AFWinPhone.utils;
using System;
using System.Diagnostics;
using System.Text;
using Windows.UI.Xaml.Controls;

namespace AFWinPhone.components.parts.validators
{
    /// <summary>
    /// Validates MIN rule, which checks if value in field is more or equal than value specified in rule.
    /// </summary>
    class MinValueValidator : AFValidator
    {
        public bool validate(AFField field, StringBuilder errorMsgs, AFValidationRule rule)
        {
            bool validationIsFine = true;
            if (Utils.IsFieldNumberField(field))
            {
                TextBox numberField = (TextBox) field.getFieldView();
                try
                {
                    if (String.IsNullOrEmpty(numberField.Text) &&
                        Convert.ToDouble(numberField.Text) < Convert.ToDouble(rule.getValue()))
                    {
                        validationIsFine = false;
                    }
                }
                catch (Exception ex) //catch convert exception
                {
                    validationIsFine = true; //if there is no number in field we cant validate its value
                    Debug.WriteLine(ex.StackTrace);
                }
            }
            if (!validationIsFine)
            {
                errorMsgs.Append(Localization.translate("validation.minval") + " " + rule.getValue());
            }
            return validationIsFine;
        }
    }
}
